lib/open_api_spex/cast/discriminator.ex

```diff
@@ -40,11 +40,11 @@ defmodule OpenApiSpex.Cast.Discriminator do
   defp cast_discriminator(%_{value: value, schema: schema} = ctx) do
     {discriminator_property, mappings} = discriminator_details(schema)
 
-    case Map.pop(value, "#{discriminator_property}") do
-      {"", _} ->
+    case value["#{discriminator_property}"] || value[:"#{discriminator_property}"] do
```

Comment:
`discriminator_property` is an atom here. No need to wrap it in `:"#{...}"` right?

```suggestion
    case value["#{discriminator_property}"] || value[discriminator_property] do
```